src/component/handlers/edit/editOnInput.js

```diff
@@ -44,6 +46,13 @@ function editOnInput(editor: DraftEditor): void {
     editor._pendingStateFromBeforeInput = undefined;
   }
 
+  // Most browsers fire a selection event before making spellcheck
```

Comment:
This comment seems to imply we're expecting the event to fire after the correction is made but in Slack it sounds like the event isn't firing at the point where the user right clicks on the word.